### ISSUE TITLE
[2014 Player] Videos watched with 2014 player aren't updating watch history

### ISSUE THREAD

**Issue by i-lost-my-bagel** [Rehike/Rehike#493]
### Rehike version

0.7

### Operating system

Linux 6.1.0-17-amd64 x86_64

### PHP version

PHP 8.2.7

### What's going on?

If I have the experimental 2014 player set then any videos watched are not added to my watch history

**Comment by OleExploder** [comment]
Same thing happened to me but in Windows 11 22H2 and when using version 0.8.1.

**Comment by TorutheRedFox** [comment]
accidental privacy W?

**Comment by olioliluaaa** [comment]
Same issue with me, Windows 10 Pro 22H2 with PHP 8.2.12, on the classic 2015 player. On the latest player it updates as usual though.

**Comment by kawapure** [comment]
> Same issue with me, Windows 10 Pro 22H2 with PHP 8.2.12, on the classic 2015 player. On the latest player it updates as usual though.

It affects everyone. The player itself simply doesn't make the requests necessary to log the watch history correctly.

**Comment by TorutheRedFox** [comment]
> > Same issue with me, Windows 10 Pro 22H2 with PHP 8.2.12, on the classic 2015 player. On the latest player it updates as usual though.
> 
> It affects everyone. The player itself simply doesn't make the requests necessary to log the watch history correctly.

actually thinking about it, i think we patched those requests out because they were causing a self-DoS as the actual endpoint itself wasn't implemented and falling back to trying to load a channel page

**Comment by kawapure** [comment]
> > > Same issue with me, Windows 10 Pro 22H2 with PHP 8.2.12, on the classic 2015 player. On the latest player it updates as usual though.
> > 
> > 
> > It affects everyone. The player itself simply doesn't make the requests necessary to log the watch history correctly.
> 
> actually thinking about it, i think we patched those requests out because they were causing a self-DoS as the actual endpoint itself wasn't implemented and falling back to trying to load a channel page

No, those were generate_204 requests which were optimised to be ignored by the server. As far as I'm aware, generate_204 has no effect on behaviour at all.

Watch history updates are controlled by a separate request (/api/stats/playback) which is requested by modern player + [in Rehike's PHP code manually here, since we needed to populate watch history for signed out sessions to have homepage recommendations.](https://github.com/Rehike/Rehike/blob/master/modules/Rehike/Boot/Tasks.php#L201) It's just an oversight in the 2014 player that prevents it from making these requests.
